src/chromap.cc

```diff
@@ -1327,22 +1329,35 @@ void Chromap<MappingRecord>::MapPairedEndReads() {
           //  }
           //}
           // Update cache
-          for (uint32_t pair_index = 0; pair_index < num_loaded_pairs / 2; ++pair_index) {
-            if (num_reads_ >= 2 * 5000000 && pair_index >= num_loaded_pairs / num_threads_) {
+          for (uint32_t pair_index = 0; pair_index < num_loaded_pairs; ++pair_index) {
+            if (num_reads_ > 5000000 && pair_index >= num_loaded_pairs / num_threads_) {
               break;
             }
             mm_to_candidates_cache.Update(mm_history1[pair_index].minimizers, mm_history1[pair_index].positive_candidates, mm_history1[pair_index].negative_candidates, mm_history1[pair_index].repetitive_seed_length);
             mm_to_candidates_cache.Update(mm_history2[pair_index].minimizers, mm_history2[pair_index].positive_candidates, mm_history2[pair_index].negative_candidates, mm_history2[pair_index].repetitive_seed_length);
-            if (mm_history1[pair_index].positive_candidates.size() < mm_history1[pair_index].positive_candidates.capacity() / 2) {
+            /*if (mm_history1[pair_index].positive_candidates.size() < mm_history1[pair_index].positive_candidates.capacity() / 50) {
+              std::vector<Candidate>().swap(mm_history1[pair_index].positive_candidates);
+            }
+            if (mm_history1[pair_index].negative_candidates.size() < mm_history1[pair_index].negative_candidates.capacity() / 50) {
+              std::vector<Candidate>().swap(mm_history1[pair_index].negative_candidates);
+            }
+            if (mm_history2[pair_index].positive_candidates.size() < mm_history2[pair_index].positive_candidates.capacity() / 50) {
+              std::vector<Candidate>().swap(mm_history2[pair_index].positive_candidates);
+            }
+            if (mm_history2[pair_index].negative_candidates.size() < mm_history2[pair_index].negative_candidates.capacity() / 50) {
+              std::vector<Candidate>().swap(mm_history2[pair_index].negative_candidates);
+            }*/
+
+						if (mm_history1[pair_index].positive_candidates.size() > 50) {
               std::vector<Candidate>().swap(mm_history1[pair_index].positive_candidates);
             }
-            if (mm_history1[pair_index].negative_candidates.size() < mm_history1[pair_index].negative_candidates.capacity() / 2) {
+            if (mm_history1[pair_index].negative_candidates.size() > 50) {
```

Comment:
Can you clarify what these swaps do and why use 50 as a threshold?

Reply:
This is used to release the memory of the candidate list. I think 50 is roughly a good number. It does not release too often and also avoids the case with too many candidates which wastes the memory.